Mathlib/RingTheory/Ideal/Maps.lean

```diff
@@ -59,38 +59,44 @@ theorem mem_map_of_mem (f : F) {I : Ideal R} {x : R} (h : x ∈ I) : f x ∈ map
 theorem apply_coe_mem_map (f : F) (I : Ideal R) (x : I) : f x ∈ I.map f :=
   mem_map_of_mem f x.2
 
-theorem map_le_iff_le_comap : map f I ≤ K ↔ I ≤ comap f K :=
+theorem map_le_iff_le_comap [RingHomClass F R S] : map f I ≤ K ↔ I ≤ comap f K :=
```

Comment:
I don't understand this change. There is `[RingHomClass F R S]` in scope, and the statement uses `f : F` and `I : Ideal R` and `K : Ideal S`, so normally `[RingHomClass F R S]` should be pulled in by the new algorithm. If it isn't, it is probably a bug, that should be fixed instead of adding the assumption (unless I'm misunderstanding something). 

Same thing for the next statements.

Reply:
Good catch. This was working around an earlier bug that is now fixed in 4814. (Not pulling in things mentioned via the type of something pulled in.)